NAS2D/Resource/Image.cpp

```diff
@@ -39,13 +39,13 @@ namespace
 }
 
 
-Image::Image(const std::string& filePath) :
-	mResourceName{filePath}
+Image::Image(const std::string& filePathOrName) :
+	mFilePathOrName{filePathOrName}
```

Comment:
I'm very hesitant to start putting "Or" in variable names. In this case, I much prefer the more generic term `mResourceName` over `mFilePathOrName`. The fact that name was once a file path actually has very little relevance once the data is loaded. At that point, it's mostly just an identifier to help with debugging.

Reply:
This is partially a consequence of the Texture/Image problem. Images have paths, textures have names/IDs.